cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/CreateMoneyFlow.kt

```diff
@@ -0,0 +1,56 @@
+package com.example.flow
+
+import co.paralleluniverse.fibers.Suspendable
+import com.example.contract.CashContract
+import com.example.state.CashState
+import com.example.utils.bankProvider
+import net.corda.core.contracts.Command
+import net.corda.core.contracts.requireThat
+import net.corda.core.flows.*
+import net.corda.core.identity.Party
+import net.corda.core.transactions.SignedTransaction
+import net.corda.core.transactions.TransactionBuilder
+
+object CreateMoneyFlow {
+    @InitiatingFlow
+    @StartableByRPC
+    class Initiator(val cashValue: Long, val moneyOwner: Party) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val notary = serviceHub.networkMapCache.notaryIdentities[0]
+            val bank = serviceHub.bankProvider()
+
+            require(ourIdentity.owningKey == bank.owningKey) { "only bank can create money" }
+
+            val cashState = CashState(cashValue, bank, moneyOwner);
+            val cashCommand = Command(CashContract.Commands.Create(), bank.owningKey)
+
+            val txBuilder = TransactionBuilder(notary)
+                    .addOutputState(cashState, CashContract.ID)
+                    .addCommand(cashCommand)
+
+            txBuilder.verify(serviceHub)
+            val signedTx = serviceHub.signInitialTransaction(txBuilder)
+            val ownerSession = initiateFlow(moneyOwner)
+
+            //return subFlow(FinalityFlow(signedTx, Collections.emptyList()))
+            return subFlow(FinalityFlow(signedTx, setOf(ownerSession)))
+        }
+    }
+
+    @InitiatedBy(CreateMoneyFlow.Initiator::class)
+    class Acceptor(val bankSession: FlowSession) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            val signTransactionFlow = object : SignTransactionFlow(bankSession) {
```

Comment:
Your `CashContract` states that only the `bank` is a required signer; which you already achieved when you called `val signedTx = serviceHub.signInitialTransaction(txBuilder)` in the initiator.  
The responder flow is only required so that money `owner` calls `ReceiveFinalityFlow` to finalize the transaction on their end.  
The reason your testing is failing is because `Acceptor` is calling `SingTransactionFlow` which in turn has a `receive()` call inside of it to receive the partially signed transaction; but since your `Initiator` doesn't `send` that transaction (by calling `CollectSignaturesFlow`), this leads to that your `Acceptor` flow is hanging (its `receive` call is waiting to receive something).  
Remove this entire block; only keep `ReceiveFinalityFlow()`.

Reply:
thank you!

Reply:
My pleasure.

Reply:
> Your `CashContract` states that only the `bank` is a required signer; which you already achieved when you called `val signedTx = serviceHub.signInitialTransaction(txBuilder)` in the initiator.

I disagree a bit and want to make sure you do not conflate 2 things:

1. the contract verifies that the expected required signers are mentioned. The contract does not have access to the signatures.
2. the Corda system, as part of the finalisation will check that the required signers have signed.

So when you do `val signedTx = serviceHub.signInitialTransaction(txBuilder)` you satisfy point 2. You satisfy point 1 when you add the command with the correct required signers.